SCF/SCF-session2/Assignment1/JobScheduler.java

```
@@ -0,0 +1,264 @@
+		
+		//input validation.
+		boolean isValidInput = inputValidation(process, noOfProcess);
+		if( ! isValidInput)
+		{
+			throw new IOException("Please enter valid inputs!");			
+		}
+		
+		//calling getWaitingTime method to get waiting time of processes.
+		int[] waitingTime = getWaitingTime(process, noOfProcess);
+		
+		int maximumWaitingTime = 0;
+		
+		//calculating maximum waiting time.
+		for(int index = 0; index < noOfProcess; index++)
+		{
+			if(waitingTime[index] > maximumWaitingTime)
+			{
+				maximumWaitingTime = waitingTime[index];
+			}
+		}
+		
+		return maximumWaitingTime;		
+	}
+	
+	/**
+	 * 
+	 * This method will sort the array by their arrival time.
+	 * @see {@link https://codippa.com/how-to-sort-a-two-dimensional-array-on-column-values-in-java/}
+	 * @param process : two dimension integer array which contain arrival time and burst time of n process.
+	 * @param noOfProcess : total no. of process.
+	 * @return 2D array which contain sorted form of process by their arrival time.
+	 */
+	public static int[][] sortProcessByArrivalTime(final int[][] process, int noOfProcess)
+	{
+		int[][] sortedProcess = process;
+		Arrays.sort(sortedProcess, new Comparator<int[]>()
+				{    
+	        @Override              
+	        public int compare(final int[] entry1, final int[] entry2)
+	        { 
+	            if(entry1[0] >= entry2[0])
+	            {
+	            	return 1;
+	            } 
+	            else
+	            {
+	            	return -1;
+	            } 
+	        } 
+	        });
+		
+		return sortedProcess;
+	}
+	
+	/**
+	 * 
+	 * @param process take a two dimension integer array containing arrival time and burst time of n process
+	 * @param numberOfProcess take a integer value containing number of Process
+	 * @return true if number of process is greater than 0 and arrival time and burst time of each process is not less than 0 and 1 else return false
+	 */
+	public static boolean inputValidation(int process[][], int numberOfProcess)
```

should be called immediately after the input is provided - fail fast methodology